File: algorithms/3_hard/T/00982_triples_with_bitwise_and_equal_to_zero/hashmap.cpp
class Solution {
public:
    int countTriplets(vector<int>& nums) {
        auto hashmap = unordered_map<int, int>();
        for (const auto n1 : nums) {
            for (const auto n2 : nums) {
                hashmap[(n1 & n2)] += 1;
            }
        }

        int cnt = 0;
        for (const int num : nums) {
            for (const auto [other, freq] : hashmap) {
                if ((num & other) == 0) {
                    cnt += freq;
                }
            }
        }

        return cnt;
    }
};

File: algorithms/3_hard/T/00982_triples_with_bitwise_and_equal_to_zero/hashmap.rs
use std::collections::HashMap;

impl Solution {
    pub fn count_triplets(nums: Vec<i32>) -> i32 {
        let mut hashmap = HashMap::new();
        for &n1 in &nums {
            for &n2 in &nums {
                *hashmap.entry(n1 & n2).or_insert(0) += 1;
            }
        }

        let mut cnt = 0_i32;
        for num in nums {
            for (&other, &freq) in &hashmap {
                if num & other == 0 {
                    cnt += freq;
                }
            }
        }

        cnt
    }
}
